### PR TITLE
[mlir][spirv] Add integration test for `vector.interleave` and `vector.shuffle`

### PR DESCRIPTION
**Pull request by obtuseangleAZ**
- Add integration test for  `vector.shuffle` and `vector.interleave`, mentioned in issue #91978
- Add `VectorToSPIRV` patterns to `GPUToSPIRVPass`

### CODE REVIEW

File: mlir/test/mlir-vulkan-runner/vector-interleave.mlir [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // Allocate 3 buffers.
+    %buf0 = memref.alloc() : memref<2xi32>
+    %buf1 = memref.alloc() : memref<2xi32>
+    %buf2 = memref.alloc() : memref<4xi32>
+    
+    %idx0 = arith.constant 0 : index
+    %idx1 = arith.constant 1 : index
+    %idx4 = arith.constant 4 : index
+
+    // Initialize input buffer
```
Comment (kuhar) [inline mlir/test/mlir-vulkan-runner/vector-interleave.mlir:33]:
```suggestion
    // Initialize input buffer.
```

File: mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include "mlir/Conversion/ArithToSPIRV/ArithToSPIRV.h"
 #include "mlir/Conversion/FuncToSPIRV/FuncToSPIRV.h"
 #include "mlir/Conversion/GPUToSPIRV/GPUToSPIRV.h"
+#include "mlir/Conversion/VectorToSPIRV/VectorToSPIRV.h"
```
Comment (kuhar) [inline mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:19]: Does this require cmake/bazel dependencies to be updated?

**Review by kuhar — COMMENTED** [review]
Overall looks fine. Just need to format the code and confirm the dependencies.

File: mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp (hunk 2 above)
Comment (angelz913) [reply mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:19]: I didn't update cmake/bazel, and didn't run into problem when rebuilding

Comment (kuhar) [reply mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:19]: These are often 'masked' with cmake -- they happen to work until you switch build options to shared libs or build with bazel. You can confirm by running the missing deps ninja tool: https://discourse.llvm.org/t/ninja-can-now-check-for-missing-cmake-dependencies-on-generated-files/74344

Comment (angelz913) [reply mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp:19]: I ran the ninja tool and did not see this file in the output

**Comment by kuhar** [comment]
@angelz913 please format the code and this should be good to land

**Review by kuhar — APPROVED** [review]
LGTM

**Comment by joker-eph** [comment]
This broke the gcc-7 bot unfortunately, I reverted in https://github.com/llvm/llvm-project/pull/93732

https://lab.llvm.org/buildbot/#/builders/264/builds/11077

**Comment by joker-eph** [comment]
Actually this crashed with clang as well: https://lab.llvm.org/buildbot/#/builders/61

**Comment by kuhar** [comment]
Thanks, @joker-eph

**Comment by kuhar** [comment]
This is most likely caused by the nvidia driver rejecting the spir-v blob, whereas amdvlk/swiftshader seem more forgiving: https://gist.github.com/kuhar/f65f36d01e8c0b6018e2344ccbf39646

**Pull request merged**